File: C++/442. Find All Duplicates in an Array.cpp
class Solution {
public:
    vector<int> findDuplicates(vector<int>& nums) {
        vector<int> ans;
        for(auto num: nums)
        {
            int v = abs(num);
            if(nums[v-1] < 0)
                ans.push_back(v);
            else
                nums[v-1] = -nums[v-1];
        }
        return ans;
    }
};

File: Python/442. Find All Duplicates in an Array.py
class Solution:
    def findDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        ans = list()
        for num in nums:
            v = abs(num)
            if nums[v-1] < 0:
                ans.append(v)
            else:
                nums[v-1] = -nums[v-1]
        return ans